v2/internal/parse_test.go

```diff
@@ -28,23 +33,180 @@ func TestParseMessageFileBytes(t *testing.T) {
+				Messages: []*Message{{
+					ID:    "nested.hello",
+					Other: "world",
+				}},
+			},
+		},
+		{
+			name: "basic test with description",
+			file: `{"notnested": {"description": "world"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:          "notnested",
+					Description: "world",
+				}},
+			},
+		},
+		{
+			name: "basic test with id",
+			file: `{"key": {"id": "forced.id"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID: "forced.id",
+				}},
+			},
+		},
+		{
+			name: "basic test with description and dummy",
+			file: `{"notnested": {"description": "world", "dummy": "nothing"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:          "notnested",
+					Description: "world",
+				}},
+			},
+		},
+		{
+			name: "deeply nested test",
+			file: `{"outer": {"nested": {"inner": "value"}}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:    "outer.nested.inner",
+					Other: "value",
+				}},
+			},
+		},
+		{
+			name: "multiple nested test",
+			file: `{"nested": {"hello": "world", "bye": "all"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:    "nested.hello",
+					Other: "world",
+				}, {
+					ID:    "nested.bye",
+					Other: "all",
+				}},
+			},
+		},
+		{
+			name: "YAML nested test",
+			file: `
+outer:
+    nested:
+        inner: "value"`,
+			path:           "en.yaml",
+			unmarshalFuncs: map[string]UnmarshalFunc{"yaml": yaml.Unmarshal},
+			messageFile: &MessageFile{
+				Path:   "en.yaml",
+				Tag:    language.English,
+				Format: "yaml",
+				Messages: []*Message{{
+					ID:    "outer.nested.inner",
+					Other: "value",
+				}},
+			},
+		},
 	}
 	for _, testCase := range testCases {
 		actual, err := ParseMessageFileBytes([]byte(testCase.file), testCase.path, testCase.unmarshalFuncs)
-		if err != testCase.err {
-			t.Fatalf("expected error %#v; got %#v", testCase.err, err)
+		if (err == nil && testCase.err != nil) ||
+			(err != nil && testCase.err == nil) ||
+			(err != nil && testCase.err != nil && err.Error() != testCase.err.Error()) {
+			t.Errorf("%s failed: expected error %#v; got %#v", testCase.name, testCase.err, err)
+			continue
+		}
+		if actual == nil {
+			continue
 		}
 		if actual.Path != testCase.messageFile.Path {
-			t.Fatalf("expected path %q; got %q", testCase.messageFile.Path, actual.Path)
+			t.Errorf("%s failed: expected path %q; got %q", testCase.name, testCase.messageFile.Path, actual.Path)
```

Comment:
Instead of prepending the test name to each error, it would be worth it to use t.Run for subtests.

Examples: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/nicksnyder/go-i18n%24+t.Run

Reply:
I hesitated when I introduced the `name` attribute, but did not want to change test structure ;-)
I also prefer to use subtests, fixed in commit https://github.com/nicksnyder/go-i18n/pull/157/commits/f0d608c5543d2c116ed88c48b7839158da3b5737